### ISSUE TITLE
Text and Textarea question types

### ISSUE THREAD

**Issue by baranan** [ProjectImplicit/PIquest#142]
1. At http://projectimplicit.github.io/PIquest/0.0/quest/API.html clicking on Text & Textarea doesn't do anything.

2. It is unclear from the documentation why we need maxlengthLimit and maxlength. Why wouldn't it be enough to set maxlength to 0 by default and any number > 0 means that we limit the length of the message?

**Comment by eladzlot** [comment]
1. fixed
2. They have unrelated roles. `maxlengthLimit` is not part of the validation scheme. It does not let you type any additional characters. `maxlength` is a validator - it warns you if you insert to many characters. - How would you phrase the documentation differently?

**Comment by baranan** [comment]
2. I figured that out when I realized that maxLength did not limit the
typing. Perhaps it should limit typing (i.e., maxlengthLimit=true) by
default. Then, you can use this information in the documentation:
(true or false; default is true) Set to false in order to allow users to
*type* more characters than the characters allowed in maxlength. In that
case, only after they try to submit the page, they will see a message
prompting them to shorten their text. (if maxlength is not defined, any
number of characters will be allowed).

**Comment by eladzlot** [comment]
It seems un-intuitive to me that all validators (including minlength) work
in one way, and only maxlength will actively prevent invalid responses. In
particular users may be surprised by their inability to commit more than
maxlength characters.
Why should users need to activate an additional flag to get standard
behaviour?

**Comment by eladzlot** [comment]
How about this instead:

> (true or false; default is false) Set to true in order to prevent users from *typing* more characters than the characters allowed by `maxlength` (if maxlength is not defined, any number of characters will be allowed). This option will not allow users to input text that will activate maxlength and show its validation error.

**Comment by baranan** [comment]
I don't think that it is the expected behavior. It is very annoying for the
participant to type a lot and only then realize that it is too long. There
is very little reason that researchers would want to allow their
participants type and type and find out that they typed too much only when
they try to submit.
